backend/src/git.rs

```diff
@@ -466,9 +710,47 @@ impl Interface {
     /// Returns the latest commit from `HEAD`.
     ///
     /// <https://zsiciarz.github.io/24daysofrust/book/vol2/day16.html>
-    fn find_last_commit(repo: &Repository) -> Result<git2::Commit, git2::Error> {
+    pub fn find_last_commit(repo: &Repository) -> Result<git2::Commit, git2::Error> {
         let obj = repo.head()?.resolve()?.peel(git2::ObjectType::Commit)?;
         obj.into_commit()
             .map_err(|_| git2::Error::from_str("Couldn't find commit"))
     }
+
+    /// Resets the working directory to the last committed state, discarding any uncommitted changes.
+    /// 
+    /// This function checks the status of the repository and, if any uncommitted 
+    /// changes are detected, it resets the working directory to the last committed state. 
+    /// This ensures that the working directory is clean before pulling changes from the remote.
+    /// 
+    /// This function is equivalent to running `git reset --hard`.
+    /// 
+    /// # Errors
+    /// This function will return an error if retrieving the status fails or if 
+    /// checking out the head fails.
+    fn git_reset(&self, repo: &git2::Repository) -> Result<()> {
+        // Get the current status of the repository
+        let status = repo.statuses(None)?;
+
+        // Log the status of each file
+        for entry in status.iter() {
+            debug!("File: {:?}, Status: {:?}", entry.path(), entry.status());
+        }
+
+        // Check for uncommitted changes
+        if status.iter().any(|s| s.status() != git2::Status::CURRENT) {
+            info!("Uncommitted changes found. Discarding changes before pulling.");
```

Comment:
This could probably be a warning, because if this happened, something went wrong but we were able to recover.
```suggestion
            warn!("Uncommitted changes found. Discarding changes before pulling.");
```